web/src/providers/Span/Span.provider.tsx

```diff
@@ -0,0 +1,81 @@
+import {noop} from 'lodash';
+import {createContext, useCallback, useContext, useMemo} from 'react';
+import {useSelector} from 'react-redux';
+import {useAppDispatch} from 'redux/hooks';
+import {setAffectedSpans, setFocusedSpan, clearAffectedSpans, setSelectedSpan} from 'redux/slices/Span.slice';
+import SpanSelectors from 'selectors/Span.selectors';
+import {TSpan} from 'types/Span.types';
+
+interface IContext {
+  selectedSpan?: TSpan;
+  affectedSpans: string[];
+  focusedSpan: string;
+  onSelectSpan(spanId: TSpan): void;
+  onSetAffectedSpans(spanIdList: string[]): void;
+  onSetFocusedSpan(spanId: string): void;
+  onClearAffectedSpans(): void;
+}
+
+export const Context = createContext<IContext>({
+  affectedSpans: [],
+  focusedSpan: '',
+  onSelectSpan: noop,
+  onSetFocusedSpan: noop,
+  onClearAffectedSpans: noop,
+  onSetAffectedSpans: noop,
+});
+
+interface IProps {
+  children: React.ReactNode;
+}
+
+export const useSpan = () => useContext(Context);
+
+const SpanProvider = ({children}: IProps) => {
+  const dispatch = useAppDispatch();
+  const selectedSpan = useSelector(SpanSelectors.selectSelectedSpan);
+  const affectedSpans = useSelector(SpanSelectors.selectAffectedSpans);
+  const focusedSpan = useSelector(SpanSelectors.selectFocusedSpan);
+
+  const onSelectSpan = useCallback(
+    (span: TSpan) => {
```

Comment:
I know you guys disagree. 
But I would abstract each hook in here to a separate file, in order to unit test them individually